internal/testrunner/runners/system/test_config.go

```diff
@@ -129,6 +135,11 @@ func newConfig(configFilePath string, svcInfo servicedeployer.ServiceInfo, servi
 		c.Agent.PreStartScript.Language = agentdeployer.DefaultAgentProgrammingLanguage
 	}
 
+	// Not included in package-spec validation for deployer name
+	if c.Deployer != "" && !slices.Contains(allowedDeployerNames, c.Deployer) {
+		return nil, fmt.Errorf("invalid deployer name %q in system test configuration file %q, allowed values are: %s", c.Deployer, configFilePath, strings.Join(allowedDeployerNames, ", "))
+	}
```

Comment:
Added validation here instead of package-spec. In package-spec this file has `additionalProperties: true`